apps/acqdat_core/lib/acqdat_core/dashboard_management/schema/panel.ex

```diff
@@ -31,6 +31,7 @@ defmodule AcqdatCore.DashboardManagement.Schema.Panel do
     # associations
     belongs_to(:org, Organisation, on_replace: :delete)
     belongs_to(:dashboard, Dashboard, on_replace: :delete)
+    belongs_to(:parent, Panel, on_replace: :delete)
```

Comment:
We should add the `has_many` relation to simplify our code. IMO, following is desirable:
1. panel cannot be created along with subpanels in a single query.
2. subpanels (`ps`) of panel (`p`) cannot be edited/deleted while editing `p`. The `children` key should never be `cast`ed.

However, if (1) is desired, then `children` key can be `cast`ed in `create_changeset` but not in `update_changeset` -- this will not conflict with (2).

```suggestion
    belongs_to(:parent, Panel, on_replace: :mark_as_invalid)  # prevents subpanels from auto-deletion. subpanels should be deleted explicitly even by raw changesets.
    has_many(:children, Panel)
```

This should help simplify queries and also reduce duplication, see other comments.